core/src/main/java/org/infinispan/health/impl/ClusterHealthImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.infinispan.health.impl;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.infinispan.health.ClusterHealth;
+import org.infinispan.health.HealthStatus;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+public class ClusterHealthImpl implements ClusterHealth {
```

Comment:
I'm not sure it if our responsibility to aggregate the cluster status with RPCs... Maybe a sophisticated monitoring tool (Zabbix, Hawkular, Splunk) should be used in this case?

The bottom line is that I would prefer to keep this API short for start. And if there are such needs, enhance it in the future.